### ISSUE TITLE
request for password never mentions the password

### ISSUE THREAD

**Issue by carandraug** [OfflineIMAP/offlineimap#558]
When logging by typing password manually, the interface is:

```
Account sync ACCOUNT-NAME:
 *** Processing account ACCOUNT-NAME
 Establishing connection to REMOTE-HOST:REMOTE-PORT (REMOTE-NAME)
Enter password for account 'REMOTE-NAME':
```

but it never actually shows the username.  If there is any issue on the username, the following is is displayed which again never shows the username

````
Account sync ACCOUNT-NAME:
 *** Processing account ACCOUNT-NAME
 Establishing connection to REMOTE-HOST:REMOTE-PORT (REMOTE-NAME)
Enter password for account 'REMOTE-NAME': 
 PLAIN authentication failed: AUTHENTICATE failed.
 LOGIN authentication failed: LOGIN failed.
 ERROR: All authentication types failed:
	PLAIN: AUTHENTICATE failed.
	LOGIN: LOGIN failed.
 *** Finished account 'ACCOUNT-NAME' in 0:03
ERROR: Exceptions occurred during the run!
ERROR: All authentication types failed:
	PLAIN: AUTHENTICATE failed.
	LOGIN: LOGIN failed.
````

Because the username is never shown on the output, if the issue is on the username debugging is a lot harder (see issue #557 for what caused the issue to me, I had to go and modify the sources to find out what was being used).

Related issue `offlineimap.imapserver.IMAPserver.__plainhandler` should probably check if `authc` is `None`. I tried to raise a `OfflineImapError` and while the exception message got printed, it continued on and asked for the password anyway.

**Comment by nicolas33** [comment]
Run offlineimap with the `-a` CLI option to know which account is failing.

**Comment by carandraug** [comment]
> Run offlineimap with the -a CLI option to know which account is failing.

I know what account is failing.  The issue is that the username used is never displayed. If there is an issue on the parsing of the config file (which was my problem) and the username is not show, it's not possible for a user to find the issue.

So I'm suggesting that instead of:

````
 Establishing connection to REMOTE-HOST:REMOTE-PORT (REMOTE-NAME)
Enter password for account 'REMOTE-NAME':
````

Something like this should be show:

```
 Establishing connection to REMOTE-HOST:REMOTE-PORT (REMOTE-NAME)
Enter password for account 'REMOTE-USERNAME':
```

I tried to do this, but the getpass method is not being passed the username. I'm also unsure if there's cases where there's not really an username?

**Comment by nicolas33** [comment]
> I tried to do this, but the getpass method is not being passed the username. I'm also unsure if there's cases where there's not really an username?

The getpass() signature shall be changed, yes. Notice that a wrong parsing of the configuration file might still lead to undefined behaviour, though.

**Comment by nicolas33** [comment]
The fix will be in the release after v7.2.1. Thanks!
